tensorflow_quantum/datasets/spin_system.py

```diff
@@ -0,0 +1,240 @@
+# Copyright 2020 The TensorFlow Quantum Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Quantum datasets for several quantum many-body spin systems in the form of
+parameterized circuits"""
+from typing import List
+import os
+import numpy as np
+import sympy
+import scipy.sparse as sps
+import cirq
+
+
+class SpinSystem(object):
+    """Abstract spin system class containing information about the quantum
+    data
+    """
+
+    def __init__(self, g: float, res_e: float, fidelity: float,
+                 gs_energy: float, parameters: np.ndarray, path: str):
+        """Instantiate this spin system.
+
+        Args:
+            g: Order parameter
+            res_e: Residual energy between the exact diagonalization ground
+            state and the circuit state.
+            fidelity: Fidelity between the exact diagonalization ground state
+            and the circuit state.
+            gs_energy: Exact diagonalization  round state energy
+            parameters: Numpy array of shape (M,depth) where M is the required
+             number of parameters per layer.
+            path: Path to the data.
+        """
+        self.g = g
+        self.res_e = res_e
+        self.fidelity = fidelity
+        self.gs_energy = gs_energy
+        self.parameters = parameters
+        self.path = path
+
+    def __lt__(self, other):
+        return self.g < other.g
+
+    def hamiltonian(self) -> sps.csr_matrix:
+        """Load the hamiltonian into a scipy sparse CSR matrix."""
+        return sps.load_npz(self.path + '/H.npz')
+
+    def ground_state(self) -> np.ndarray:
+        """Load the ground state wave function into a complex numpy array."""
+        return np.load(self.path + '/groundstate.npy')
+
+    def final_state(self):
```

Comment:
Do we need this if a user can do cirq.final_wavefunction(the_circuit(parameters)) to get this ?

Reply:
No, probably not.